### PR TITLE
Separate RL configurations into a dedicated config class

### PR DESCRIPTION
**Pull request by SurbhiJainUSC**
# Description

This PR separates RL configurations into its own dedicated configuration class. 

**Why is this change being made & the problem being solved:**
Previously, RL-specific configurations were mixed with the pre-training configurations. As the number of RL-specific features and hyperparameters grows, this mixed approach makes the primary configuration structure bloated and harder to maintain. 

**Why this is a good solution:**
By isolating RL configurations into a dedicated class, we improve the modularity and readability of our configuration management. It enforces a clear separation of concerns, ensuring that non-RL workloads aren't cluttered with RL-specific parameters, and makes RL hyperparameters easier to track and modify independently.

**Specific implementation:**
- Created a new dedicated config class specifically for RL parameters.
- Migrated existing RL configuration variables out of the base config and into this new class.
- Updated configuration references across the codebase to point to the newly established RL config class.

# Tests

Command: https://paste.googleplex.com/5172847176712192

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by igorts-git — APPROVED** [review]

File: src/maxtext/configs/post_train/rl.yml [hunk 1]
```diff
@@ -235,19 +239,80 @@ eval_dataset_name: 'openai/gsm8k'
...
-# AND a comma-separated list of function names to import from that file.
-# Each function signature: fn(prompts, completions, tmvp_config, **kwargs) -> list[float].
-# When both are set, the built-in [match_format_exactly, match_format_approximately,
-# check_numbers] reward list is REPLACED entirely.
-reward_functions_path: ''
-reward_functions: ''
 train_split: 'train'
 eval_split: 'test'
-hf_name: 'main' # subset of Hugging Face dataset
-tokenizer_type: 'huggingface'
```
Comment (A9isha) [inline src/maxtext/configs/post_train/rl.yml:248]: we no longer need `tokenizer_type`?

**Review by A9isha — APPROVED** [review]
